hax/corrections_handler.py

```diff
@@ -116,10 +116,21 @@ def get_correction(self, correction_name, run, ismap=False):
         # If we get here something is wrong and we didn't find the correction
         raise ValueError("Didn't find a correction for %s in run %i" % (correction_name, run))
 
-    def get_electron_lifetime_correction(self, run_start, drift_time, value='DEFAULT'):
+    def get_electron_lifetime_correction(self, run_number, run_start, drift_time, value='DEFAULT'):
         """Wrapper that does the exponential calculation for you
         """
-        return np.exp((drift_time / 1e3) / self.get_electron_lifetime(run_start, value))
+        if hax.mc_data:
+            elifetime = self.get_misc_correction("mc_electron_lifetime_liquid", run_number)
+
+        else:
+            try:
+                elifetime = self.get_electron_lifetime(run_start, value)
+
+            except Exception as e:
+                print(e)
+                return 1.
```

Comment:
If there is no electron lifetime available, are you sure you want to produce a correction minitree? 

The generic except catching with a default of 1 instead of NaN might cause some fun later on. If someone puts a syntax error in `get_electron_lifetime`, everything will run happily but we'll get no electron lifetime corrections. Hopefully someone is checking the logs and they'll see there was a syntaxerror somewhere (but since it's just a print they also don't know where).

Reply:
I think @coderdj implemented this at first; @coderdj what was your motivation for this?

Reply:
I copy-pasted code from Jacques who had added this condition for MC I think (or at least that's what I thought when I copy-pasted). But we could remove it as far as I'm concerned.

Reply:
OK raising the exception instead in e3525fd

Reply:
@coderdj Pretty sure I didnt add the e-lifetime=1 option. in fact i believe git blame says you did.

Reply:
Yeah now that I look at it I did. I don't know why I did.

But the other part is true. When I did the refactor of the corrections stuff I also noticed it was weird and copy-pasted it thinking you added it for MC stuff (without checking if that was the case). 

Anyway, it can be removed.